master: nightly test failed

The nightly  test run failed on b751e8763d7ba6cd05d065e2a546d9b29c5efea2. Please review the run [here](https://github.com/cockroachdb/pebble/actions/runs/14690905678).

Jira issue: PEBBLE-427

```
--- FAIL: TestCompactionCorruption (120.16s)
    compaction_test.go:2960: testdata/compaction_corruption:47: timeout waiting for no external files
```

I need to rethink this test.

--- a/compaction_test.go
+++ b/compaction_test.go
@@ -2902,15 +2902,13 @@ func TestCompactionCorruption(t *testing.T) {
 			},
 		},
 		L0CompactionThreshold:     1,
-		L0CompactionFileThreshold: 10,
+		L0CompactionFileThreshold: 5,
 	}
 	opts.WithFSDefaults()
 	remoteStorage := remote.NewInMem()
 	opts.Experimental.RemoteStorage = remote.MakeSimpleFactory(map[remote.Locator]remote.Storage{
 		"external-locator": remoteStorage,
 	})
-	opts.EnsureDefaults()
-	opts.Levels[0].TargetFileSize = 8192
 	d, err := Open("", opts)
 	require.NoError(t, err)
 
@@ -2928,16 +2926,17 @@ func TestCompactionCorruption(t *testing.T) {
 			defer workloadWG.Done()
 			for !stopWorkload.Load() {
 				b := d.NewBatch()
-				// Write some random keys of the form a012345.
-				for i := 0; i < 100; i++ {
-					v := make([]byte, 100+rand.IntN(100))
-					for i := range v {
-						v[i] = byte(rand.Uint32())
-					}
-					key := fmt.Sprintf("%c%06d", 'a'+byte(rand.IntN(int('z'-'a'+1))), rand.IntN(1000000))
-					if err := b.Set([]byte(key), v, nil); err != nil {
-						panic(err)
-					}
+				// Write a random key of the form a012345 and flush it. This will result
+				// in (mostly) non-overlapping tables in L0.
+				var valSeed [32]byte
+				for i := range valSeed {
+					valSeed[i] = byte(rand.Uint32())
+				}
+				v := make([]byte, 1024+rand.IntN(10240))
+				_, _ = rand.NewChaCha8(valSeed).Read(v)
+				key := fmt.Sprintf("%c%06d", 'a'+byte(rand.IntN(int('z'-'a'+1))), rand.IntN(1000000))
+				if err := b.Set([]byte(key), v, nil); err != nil {
+					panic(err)
 				}
 				if err := b.Commit(nil); err != nil {
 					panic(err)

